azurelinuxagent/ga/update.py

```diff
@@ -982,6 +986,34 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    @staticmethod
+    def _add_dns_tcp_firewall_rule_if_not_enabled(dst_ip):
+
+        # Helper to execute a run command, returns True if no exception else returns False
+        def _execute_run_command(command):
+            try:
+                shellutil.run_command(command)
+                return True
+            except Exception:
+                return False
+
+        # "-C" checks if the iptable rule is available in the chain. It throws an exception if the ip table rule doesnt exist
+        drop_rule = _get_firewall_drop_command("-w", "-C", dst_ip)
+        if not _execute_run_command(drop_rule):
+            return
+        else:
+            accept_non_root = _get_firewall_accept_dns_tcp_request_command("-w", "-C", dst_ip)
+            if not _execute_run_command(accept_non_root):
+                try:
+                    logger.info("Firewall rule to allow DNS TCP request to wireserver for a non root user unavailable . Setting it now.")
+                    accept_non_root = _get_firewall_accept_dns_tcp_request_command("-w", "-I", dst_ip)
+                    shellutil.run_command(accept_non_root)
+                    logger.info("Succesfully added firewall rule to allow non root users to do a DNS TCP request to wireserver ")
+                except Exception as e:
+                    msg = "Unable to set the nonroot tcp access firewall rule:{0}".format(ustr(e))
+                    logger.error(msg)
+
+
```

Comment:
NIT: Extra line here. Throwing a PEP violation warning - `PEP 8: E303 too many blank lines (3)`

Reply:
Addressed in the next commit